### ISSUE TITLE
`conda update anaconda` still shows Python version as Anaconda 2.3.0

### ISSUE THREAD

**Issue by ijstokes** [ContinuumIO/anaconda-issues#984]
I think something weird is going on with package pinning, but I can't really tell.  Basically I want to update to Anaconda 4.1.  I tought `conda update anaconda` would do that, but now I'm pretty sure it doesn't, and `conda install anaconda=4.1.0` gave me issues with `spyder` and `gevent`.

I would expect that `conda` would tell me if `anaconda` was pinned when I try to do an update -- is that a reasonable expectation?  Should that have happened?  (I think this might be a question for @mcg1969 or someone from the Conda team)

I think this shows my initial experience and how I was surprised by the result (still appearing to have Anaconda 2.3.0).

```bash
➜  ~ conda update anaconda
Fetching package metadata .........
Solving package specifications: ..........

Package plan for installation in environment /Users/ijstokes/anaconda:

The following packages will be DOWNGRADED due to dependency conflicts:

    openssl:    1.0.2h-1      --> 1.0.1k-1     
    pip:        8.1.2-py27_0  --> 7.0.3-py27_0 
    python:     2.7.12-1      --> 2.7.10-0     
    setuptools: 23.0.0-py27_0 --> 17.1.1-py27_0
    sqlite:     3.13.0-0      --> 3.8.4.1-1    
    zlib:       1.2.8-3       --> 1.2.8-0      

Proceed ([y]/n)? y

Unlinking packages ...
[      COMPLETE      ]|###################################################| 100%
Linking packages ...
[      COMPLETE      ]|###################################################| 100%

$ python -V
Python 2.7.10 :: Anaconda 2.3.0 (x86_64)
```

My expectation was that this would get me `Anaconda 4.1.0` from the `python -V` command.

**Comment by ijstokes** [comment]
After following the procedure documented in https://github.com/Anaconda-Platform/nb_conda_kernels/issues/44 I was able to get Anaconda 4.1.1 installed:

```bash
$ python -V
Python 2.7.12 :: Anaconda 4.1.1 (x86_64)
```

**Comment by mcg1969** [comment]
This doesn't strike me as a bug, though it is certainly an example of where conda's messaging needs to improve. `conda update anaconda` does not give you the latest version of `anaconda`; it gives you the latest version _that is compatible with your installed packages_.

I'd say it would be a reasonable, and even important, improvement to conda for it to say: "the latest version of `anaconda` is 4.1.0, but due to dependency conflicts the best we could do is `2.3.0`."

**Comment by mcg1969** [comment]
Well, let me qualify here: I'm not familiar with how pinning affects the update logic. My guess about how it _should_ work is that the pinned package is appended to the specs that the solver sees---so that the only package that meets the augmented specs is the pinned one. If that's how it is being done, then this behavior makes some sense.

It's important to point out that anaconda itself need not be the one that's pinned. If you've pinned one of anaconda's _dependencies_, then that will effectively pin anaconda as well.

**Comment by ijstokes** [comment]
@csoja I was just working with Atish and ran into this problem again.  Atish had Anaconda 2.3.  He had "old launcher".  I verbally gave him some instructions on how to update to make things work, but they didn't get him anywhere.  We had to do a  screenshare which is when I discovered that he was thoroughly stuck on Anaconda 2.3.  It took my expert intervention to guide him on a  path that allowed him to get onto Anaconda 4.1.1, and even then I was only able to do it because just weeks ago I had run into almost the same problem myself.  I would like to consider myself "not a Conda nor Anaconda novice".  Here is the issue:

1. Someone like Atish is a major part of our target audience for Anaconda: he was stuck and couldn't get himself unstuck, even with some help/suggestions from me (verbally, or the kind you'd get via an email exchange).

2. `conda update anaconda` didn't do anything, I presume because of pinning.  Atish wasn't given any hints that this wasn't updating him to the latest Anaconda (that is the theme of this issue and why I'm adding this comment).

3. `conda install anaconda=4.1.1` didn't work because of a `spyder-app` conflict.  I had to tell Atish to `conda remove spyder-app` to make progress.  Once he had done that, he was able to "upgrade".

I believe many people will have been stuck in this same scenario (gosh, I was, just a few weeks ago, and didn't even realize it or know how to proceed).  I don't think we serve those people well right now.  I think the shortest path to an upgraded system for them is to delete their current Anaconda installation and re-install the latest version.  But how awful is that! We claim to provide a flexible, modular, extensible data science platform, but by the way, if you want to upgrade, delete the old one and re-install.  And, by the way, if you `conda install`ed anything before, that is all gone.  Or otherwise, the upgrade path is some or all of:

```
conda update conda
conda remove spyder-app
conda install anaconda=4.1.1
```

But we don't tell users that (and different users might have **more** `conda remove` ops that they have to do in order to avoid conflicts), and those steps certainly aren't obvious.  Especially since `conda update anaconda` just doesn't do what the typical user would expect: update Anaconda.

**Comment by ijstokes** [comment]
@csoja I think this needs to be escalated: @rrenfrow86 and IanL both have this exact same issue.  They are stuck with old versions of Anaconda and are unable to update by any obvious means.  I believe **all** Anaconda users will be in this situation.  If not, I am interested to know what scenario would allow **anyone** to upgrade from one version of Anaconda to another.  I cannot see how that is possible today, though it may be.

**Comment by ijstokes** [comment]
I had thought one upgrade strategy was to "just install the latest version from the standard installer", however when I tried that on my Mac (granted, trying to install 4.0.0, because I wanted to test if Anaconda upgrades are, in fact, impossible) I got this message:

![screen shot 2016-09-08 at 10 07 08 am](https://cloud.githubusercontent.com/assets/293525/18352400/4ebf8ddc-75ac-11e6-8ebf-59108263e43f.png)

Did the person who created that dialog ever try that operation?  Did it work for them?  If so, that is somewhat good news: an upgrade route might be possible.  But my experience suggests it isn't possible, so I'd guess no one actually tried to do that.

**Comment by ijstokes** [comment]
Update: On Linux, the following work:

1. Virgin install of Anaconda3-4.0.0 for Linux, `conda update anaconda` works just fine.
2. From the virgin Anaconda3-4.0.0 for Linux, I updated one package (I picked `tornado`) to a version that isn't in 4.0.0 or 4.1.1 and then tried `conda update anaconda` to observe effect: `tornado` is downgraded and Anaconda (successfully) upgraded from 4.0.0 to 4.1.1
3. Virgin install of Anaconda3-4.0.0 for Mac, `conda update anaconda` works just fine and upgrades to 4.1.1
4. Modified the root environment of 4.0.0, `anaconda` becomes `custom`, then `conda update anaconda` worked just fine and shifted everything to 4.1.1 versions (upgrades and downgrades).
5. From Navigator on OS X with lightly modified 4.0.0 base system tried GUI version to upgrade `anaconda` package.  Needed to use latest Navigator (version in 4.0.0 had bug for handling channels with tokens in my `.condarc` file).  This upgrade appeared to work just fine.

So this problem is, thankfully, not universal, but the fact that myself, Atish, IanL, and @rrenfrow86 all have experienced it does suggest that it is prevalent.  I am unable to reproduce (as I've shown above), however Ryan has saved his Anaconda installations that he is unable to upgrade, so that may present a good avenue for some forensics.
